forester/src/epoch_manager.rs

```diff
@@ -29,77 +30,39 @@ use light_test_utils::indexer::{Indexer, MerkleProof, NewAddressProofWithContext
 use light_test_utils::rpc::rpc_connection::RpcConnection;
 use log::{debug, error, info, warn};
 use rand::Rng;
-use solana_account_decoder::UiAccountEncoding;
-use solana_client::nonblocking::pubsub_client::PubsubClient;
-use solana_client::rpc_config::{RpcAccountInfoConfig, RpcProgramAccountsConfig};
-use solana_sdk::commitment_config::CommitmentConfig;
 use solana_sdk::compute_budget::ComputeBudgetInstruction;
 use solana_sdk::instruction::Instruction;
 use solana_sdk::pubkey::Pubkey;
 use solana_sdk::signature::{Signature, Signer};
 use solana_sdk::transaction::Transaction;
 use std::collections::HashMap;
 use std::iter::Zip;
-use std::str::FromStr;
+use std::pin::Pin;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
-use std::time::{Duration, Instant};
-use std::{mem, thread};
-use tokio::runtime::Builder;
+use std::time::Duration;
 use tokio::sync::{mpsc, oneshot, Mutex, Semaphore};
-use tokio::time::sleep;
-
-type Result<T> = std::result::Result<T, ForesterError>;
-
-#[derive(Debug, Clone)]
-pub struct QueueItemData {
-    pub hash: [u8; 32],
-    pub index: usize,
-}
-
-pub async fn fetch_queue_item_data<R: RpcConnection>(
-    rpc: &mut R,
-    queue_pubkey: &Pubkey,
-) -> Result<Vec<QueueItemData>> {
-    debug!("Fetching queue data for {:?}", queue_pubkey);
-    let mut account = rpc.get_account(*queue_pubkey).await?.unwrap();
-    let nullifier_queue: HashSet = unsafe {
-        HashSet::from_bytes_copy(&mut account.data[8 + mem::size_of::<QueueAccount>()..])?
-    };
-    let mut queue_item_data_list = Vec::new();
-
-    for i in 0..nullifier_queue.capacity {
-        let bucket = nullifier_queue.get_bucket(i).unwrap();
-        if let Some(bucket) = bucket {
-            if bucket.sequence_number.is_none() {
-                queue_item_data_list.push(QueueItemData {
-                    hash: bucket.value_bytes(),
-                    index: i,
-                });
-            }
-        }
-    }
-
-    Ok(queue_item_data_list)
-}
-
-pub async fn get_queue_length<R: RpcConnection>(rpc: &mut R, queue_pubkey: &Pubkey) -> usize {
```

Comment:
Is this used anywhere?

Initially I wanted to comment with a question whether there is any place where we'd like to fetch the queue just for getting that number, without doing anything else, but then I couldn't find any calls of this function in the diff. 😅

Reply:
Hm, I think I already removed it from the code before your review.

Reply:
Alright, maybe I started reviewing earlier and clicked "publish" too late